### DISCUSSION TITLE
Why Deno prefers underscores to dash?

### DISCUSSION THREAD

**Discussion by Neos21** [denoland/deno#12415] [Q&A] [answered]
I used Node.js for several years. and used 'dashes (hyphens)' for the filenames.

I started learning Deno yesterday. Then I found the style guide.

> Use underscores, not dashes in filenames.
> 
> - [Style guide | Manual | Deno](https://deno.land/manual@v1.14.3/contributing/style_guide#use-underscores-not-dashes-in-filenames)

Why Deno prefers underscores to dash? I want to know the reason, background story.
Is there the problem with using dashes?

**Comment by Soremwar** [comment, marked as answer]
That is the contributing style guide. Deno itself doesn't prefer any file names

**Reply by kitsonk** [reply]
And to augment, style guides are for consistency purposes, not rigid rules for the rest of the community to follow.

**Reply by mfulton26** [reply]
I'd be interested in understanding why the _contributing_ style guide says to prefer underscores over dashes. From what I've seen most URL paths avoid underscores and JavaScript modules are typically accessed via URLs so why not use dashes?

**Reply by Neos21** [reply]
Is it because the Rust language prefers underscores to dashes?

**Reply by mfulton26** [reply]
Oh that is an interesting insight. I haven't developed in Rust personally so I had no idea that such could be a contributing factor. That sounds plausible to me.

**Comment by justinmchase** [comment]
I don't think it really matters, so long as its self-consistent and enforced by a tool.

PS - They have names [`kebab-case`](https://dencode.com/string/kebab-case?v=hello_world) and [`snake_case`](https://dencode.com/string/snake-case?v=hello-world)

**Comment by djradon** [comment]
JavaScript identifiers don't support dashes... only underscores.
